src/classes/EndOfDaySchedule.cls

```diff
@@ -0,0 +1,18 @@
+global class EndOfDaySchedule implements Schedulable {
+    
+    global void execute(SchedulableContext SC) {
+        
+        List<HTask__c> ltask = [SELECT Id 
+                                FROM HTask__c 
+                                WHERE Status__c <> 'Completed' 
+                                AND Type__c = 'Daily' 
+                                AND DueDate__c < :Date.today()];
+        
+        for(HTask__c task:ltask){            
+            Contact conta = [SELECT Id, Name FROM Contact WHERE Id = :task.Contact__c];
+            conta.Points__c -= task.Points__c;
+            
+            update conta;
```

Comment:
please refrain from executing update actions inside for loops. You should figure out a way to do that outside the loop.